### DISCUSSION TITLE
3rd party z-wave tilt sensor is showing up in MQTT/HA but without binary sensor state

### DISCUSSION THREAD

**Discussion by dredjohn21** [tsightler/ring-mqtt#449] [Support Q&A]
I added a 3rd party z-wave tilt sensor (by Zooz) to my Ring Alarm.  In Ring's app I can see the state changes when my garage door opens/closes.

However, in HAOS MQTT I'm only seeing 3 entities:
1. Tilt Sensor Tamper (e.g. Off)
2. Tilt Sensor Battery Status (e.g. 100%)
3. Tilt Sensor Info w/ Attributes:
       batteryLevel: 100
       batteryStatus: full
       commStatus: ok
       lastCommTime: 2022-09-01T23:19:01Z
       lastUpdate: 2022-09-01T23:19:01Z
       linkQuality: ok
       tamperStatus: ok
       icon: mdi:information-outline
       friendly_name: Garage Door Tilt Sensor Info

For other MQTT sensors I'm seeing a binary sensor that shows state changes when open/closed.  

I read the documentation and see that 3rd party z-wave tilt sensors are supported.  Is there something I need to do to get this to show up?  Worst case I may be able to create a binary template sensor based on the changes in time stamps, but wouldn't be very reliable as I can't tell if open/closed.

Any thoughts?

**Comment by tsightler** [comment]
Maybe a bug, I don't have a tilt sensor to actually test with, and some may report differently than others.  Can you please provide full logs and perhaps the output from device data discovery (https://github.com/dgreif/ring/wiki/Data-Discovery)?  If so I will promote this discussion to an issue and we can work on it there.

**Reply by dredjohn21** [reply]
Thanks for your help

Here are the log items related to the tilt sensor:
`022-09-02T00:50:08.827Z ring-mqtt [Garage Door Tilt Sensor] ring/dr730c-3j66v-0/alarm/247f07dc-de0c-40dd-ae96-7db8a43e65da/binary_sensor/state OFF
2022-09-02T00:50:08.828Z ring-attr [Garage Door Tilt Sensor] ring/dr730c-3j66v-0/alarm/247f07dc-de0c-40dd-ae96-7db8a43e65da/info/state {"batteryLevel":100,"batteryStatus":"full","commStatus":"ok","lastCommTime":"2022-09-02T00:08:24Z","lastUpdate":"2022-09-02T00:08:24Z","linkQuality":"ok","tamperStatus":"ok"}
2022-09-02T00:50:08.828Z ring-attr [Garage Door Tilt Sensor] ring/dr730c-3j66v-0/alarm/247f07dc-de0c-40dd-ae96-7db8a43e65da/battery/attributes {"batteryLevel":100,"batteryStatus":"full"}`

Also, I'm using HAOS and installed the SSH/Terminal add on, but when I tried to run the Device Data Discovery commands on that link it wouldn't work:
<img width="399" alt="Screen Shot 2022-09-01 at 8 53 01 PM" src="https://user-images.githubusercontent.com/95455576/188036271-027d3bff-2a6e-4993-ac43-118661423ee5.png">

Haven't tried to run any terminal commands before, so apologies if I'm doing something stupid here. :)

**Reply by tsightler** [reply]
You can run the discovery from any system with NodeJS installed, so that might be easier to install this and run on your desktop than attempting to do it from the remote CLI since there's no NodeJS installed on the docker image that you are in when you SSH into the system.

It is technically possible to do so from the HAOS CLI however, assuming the addon is running, and you've disabled protected mode in the SSH addon, then you can do something like:

`docker exec -i addon_03cabcc9_ring_mqtt /app/ring-mqtt/node_modules/ring-client-api/ring-device-data-cli.js`

Or you could even just install NodeJS temporarily (it would be gone on the next reboot) within the SSH container by doing:
```
➜  ~ apk add npm
➜  ~ npx -p ring-client-api ring-device-data-cli
```
This is a little heavy handed as it installs a bunch of stuff you only need for a one-time run, but the install is not persistent and so it will all be gone the next time the host is rebooted so it still may be the easiest way.

One thing that's clear is that the device is not being discovered as a tilt sensor, but rather just as a generic binary sensor, which is OK (this only impacts the topic name and default icon in HA, which you can always change manually anyway.  However, what's not clear is why it's not showing up any state as the data clearly shows the state topic, so it makes me think it must be a discovery bug.  It would be nice to see the logs from when the door is opened/closed.
